lib/chess_web/router.ex

```diff
@@ -35,13 +36,13 @@ defmodule ChessWeb.Router do
     pipe_through [:browser, :auth, :ensure_auth]
 
     resources "/games", GameController,
-      only: [:index, :create, :show, :delete]
+      only: [:index, :new, :create, :show, :delete]
   end
 
   # Other scopes may use custom stacks.
-  scope "/api", ChessWeb do
-    pipe_through :api
+  scope "/api", as: :api do
+    pipe_through [:api, :auth, :ensure_auth]
```

Comment:
where does `ensure_auth` come from?

Reply:
It's a pipeline defined further up in the `router.ex` file.

https://github.com/danbee/chess/blob/game-has-opponent/lib/chess_web/router.ex#L16-L18